pkg/platform/runtime/buildscript/json.go

```diff
@@ -0,0 +1,91 @@
+package buildscript
+
+import (
+	"bytes"
+	"io"
+)
+
+func (s *Script) ToJson() []byte {
+	json := &bytes.Buffer{}
+	s.WriteJson(json)
+	return json.Bytes()
+}
+
+func writeString(w io.Writer, s string) {
+	w.Write([]byte(`"`))
+	w.Write([]byte(s))
+	w.Write([]byte(`"`))
+}
+
+func (s *Script) WriteJson(w io.Writer) {
+	w.Write([]byte(`{"let":{`))
+	for i, assignment := range s.Let.Assignments {
+		assignment.WriteJson(w)
+		if i+1 < (len(s.Let.Assignments)) {
+			w.Write([]byte(","))
+		}
+	}
+	w.Write([]byte(`,"in":`))
+	switch {
+	case s.In.FuncCall != nil:
+		s.In.FuncCall.WriteJson(w)
+	case s.In.Name != nil:
+		writeString(w, "$"+*s.In.Name)
+	}
+	w.Write([]byte(`}}`))
+}
+
+func (a *Assignment) WriteJson(w io.Writer) {
+	writeString(w, a.Key)
+	w.Write([]byte(":"))
+	a.Value.WriteJson(w)
+}
+
+func (v *Value) WriteJson(w io.Writer) {
```

Comment:
Is this a pattern suggested by the participle package? If not, I suggest looking into whether ths "value" field can be an interface type where each of the potential types implement that interface. Based on my suggestions, something like:
```
type Valuer interface {
    fmt.Stringer
    json.Marshaler
}
```
Or, roughly based on existing methods:
```
type Valuer interface {
    io.WriterTo
    WriteJSONTo(io.Writer) (int, error)
}
```

Reply:
I just wanted a way to efficiently concatenate a lot of strings piecemeal, and `bytes.Buffer` coupled with an `io.Writer` seemed to fit the bill. I've implemented your `String()` and `MarshalJSON()` interface ideas, so this should not be necessary anymore.